secretariat/tests/factories.py

```diff
@@ -30,4 +30,6 @@ class Meta:
 
     user = factory.SubFactory(UserFactory)
     organisation = factory.SubFactory(OrganisationFactory)
+    start_date = Faker("date_between", start_date="-3y", end_date="+1y")
+    end_date = Faker("date_between", start_date=start_date, end_date="+3y")
```

Comment:
j'invoque la grande @tut-tuuut pour un conseil sur Faker et la end_date. 

J'ai mis "start_date=start_date" en espérant qu'il comprendrait mais je me retrouve avec des tests qui plantent dès le début et aléatoirement sur : 
```
self.assertTrue(membership.start_date < membership.end_date)
```

Est-ce que je mets start_date entre -3 an et today et end_date à partir de court-circuiter le souci ou est-ce que tu connais une manière smart de faire ça ?

Reply:
Je pense que c'est start_date=start_date à la ligne 34 qui pose problème. Pour les valeurs par défaut tu peux faire simple : 

```suggestion
    end_date = Faker("date_between", start_date="+1y", end_date="+3y")
```

Reply:
Oui, c'était fait exprès `start_date=start_date`, j'essayais de lui passer le paramètre de la ligne du dessus mais je suis pas sûre qu'il comprenne. 

Pour +1y dans les deux champs, ça peut quand même ne pas fonctionner et faire une valeur de start_date qui sera après end_date, non ?

Reply:
Ce que je comprends c'est que start_date est calée quelque part entre "il y a 3 ans" et "dans un an", et que "end_date" est calée par défaut entre "dans un an" et "dans 3 ans", et que ça devrait se passer bien sauf dans le cas où ils caleraient la start_date et la end_date exactement dans un an.

Mais comme là c'est une factory et que ça ne pose pas de problème que ces dates soient vides, je crois que je n'aurais rien mis, laissant par défaut les dates à la valeur par défaut de la BDD (null). Et ensuite lorsque j'utilise la factory pour tester des cas particuliers, je donne des dates précises : 

```python
membership_qui_doit_marcher = MembershipFactory(start_date="2023-01-01", end_date="2024-01-01")
membership_qui_ne_doit_pas_marcher = MembershipFactory(start_date="2023-01-01", end_date="2022-01-01")

Reply:
Yes, j'ai eu ces deux idées-là hier soir pendant Pathfinder. C'est plus smart de laisser null en fait.